art_internal_impl.hpp

```diff
@@ -196,13 +218,23 @@ inline void basic_db_inode_deleter<INode, Db>::operator()(
 #endif  // UNODB_DETAIL_WITH_STATS
 }
 
-template <class Db, template <class> class CriticalSectionPolicy, class NodePtr,
-          class INodeDefs, template <class> class INodeReclamator,
-          template <class, class> class LeafReclamator>
+// The basic_art_policy encapsulates differences between plain and OLC
+// ART, such as extra header field, and node access critical section
+// type.
+template <class Db,
+          template <class> class CriticalSectionPolicy,
+          class LockPolicy,
+          class ReadCriticalSection,
+          class NodePtr,
+          class INodeDefs,
+          template <class> class INodeReclamator,
+          template <class, class> class LeafReclamator
+          >
 struct basic_art_policy final {
   using node_ptr = NodePtr;
   using header_type = typename NodePtr::header_type;
-
+  using lock_policy = LockPolicy;
+  using read_critical_section = ReadCriticalSection;
```

Comment:
Is it actually used? Current implementations are immediately concrete in art.cpp / olc_art.cpp